### PR TITLE
Update Material Design links and assets to Material 3

### PR DESCRIPTION
**Pull request by rexthecoder**
fixes https://github.com/flutter/website/issues/8586

This PR Update Material Design links and assets to Material 3.

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/get-started/flutter-for/uikit-devs.md [hunk 1]
```diff
@@ -2173,8 +2173,8 @@ class _SampleAppPageState extends State<SampleAppPage> {
 [layout tutorial]: {{site.url}}/ui/widgets/layout
 [`Localizations`]: {{site.api}}/flutter/widgets/Localizations-class.html
 [Material Components]: {{site.material}}/develop/flutter/
-[Material Design guidelines]: {{site.material}}/design/
-[optimized for all platforms]: {{site.material}}/design/platform-guidance/cross-platform-adaptation.html#cross-platform-guidelines
```
Comment (guidezpl) [inline src/get-started/flutter-for/uikit-devs.md:2177]: this is one of those pages that doesn't have a counterpart in M3 (and won't) so I think we should keep linking to the m2 site

File: src/get-started/flutter-for/xamarin-forms-devs.md [hunk 2]
```diff
@@ -2573,10 +2573,10 @@ For more information on using the Firebase Cloud Messaging API, see the
 [Introduction to declarative UI]: {{site.url}}/get-started/flutter-for/declarative
 [`Localizations`]: {{site.api}}/flutter/widgets/Localizations-class.html
 [Material Components]: {{site.url}}/ui/widgets/material
-[Material Design]: {{site.material}}/design
-[Material Design guidelines]: {{site.material}}/design
+[Material Design]: {{site.material}}/styles
+[Material Design guidelines]: {{site.material}}/styles
 [`Opacity` widget]: {{site.api}}/flutter/widgets/Opacity-class.html
-[optimized for all platforms]: {{site.material}}/design/platform-guidance/cross-platform-adaptation.html#cross-platform-guidelines
```
Comment (guidezpl) [inline src/get-started/flutter-for/xamarin-forms-devs.md:2579]: same here

File: src/ui/layout/building-adaptive-apps.md [hunk 3]
```diff
@@ -1191,8 +1191,8 @@ Here are some resources to get you started:
 
 [Build high quality apps (Android)]: {{site.android-dev}}/quality
 [Human interface guidelines (Apple)]: {{site.apple-dev}}/design/human-interface-guidelines/
-[Material design for large screens]: {{site.material}}/blog/material-design-for-large-screens
+[Material design for large screens]: {{site.material2}}/blog/material-design-for-large-screens
 [Machine sizes and breakpoints (Microsoft)]: https://docs.microsoft.com/en-us/windows/uwp/design/layout/screen-sizes-and-breakpoints-for-responsive-design
-[Material guidelines on responsive UI layout]: {{site.material}}/archive/guidelines/layout/responsive-ui.html
+[Material guidelines on responsive UI layout]: {{site.material2}}/archive/guidelines/layout/responsive-ui.html
```
Comment (guidezpl) [inline src/ui/layout/building-adaptive-apps.md:1196]: There are better resources now under https://m3.material.io/foundations/layout/applying-layout and https://m3.material.io/foundations/layout/canonical-layouts/overview

File: src/ui/layout/index.md [hunk 4]
```diff
@@ -1250,9 +1250,9 @@ The following resources might help when writing layout code.
 [`ListTile`]: {{api}}/material/ListTile-class.html
 [`ListView`]: {{api}}/widgets/ListView-class.html
 [`MainAxisAlignment`]: {{api}}/rendering/MainAxisAlignment.html
-[Material card]: {{site.material}}/design/components/cards.html
-[Material Design]: {{site.material}}/design
-[Material Design palette]: {{site.material}}/design/color
+[Material card]: {{site.material}}/components/cards
+[Material Design]: {{site.material}}/styles
+[Material Design palette]: {{site.material}}/styles/color
```
Comment (guidezpl) [inline src/ui/layout/index.md:1255]: I don't think we offer a palette anymore, consider rewriting

File: src/get-started/flutter-for/uikit-devs.md (hunk 1 above)
Comment (rexthecoder) [reply src/get-started/flutter-for/uikit-devs.md:2177]: [Updated ](https://github.com/flutter/website/pull/8706/commits/465fee0a6a3c77969f77223165f6d7cfb662bec5#diff-2bd4c52d8555721f7620ab1c9c3848f7f3a6cbd59d819da34d62bf3e5a96fe48)

File: src/get-started/flutter-for/xamarin-forms-devs.md (hunk 2 above)
Comment (rexthecoder) [reply src/get-started/flutter-for/xamarin-forms-devs.md:2579]: [Updated](https://github.com/flutter/website/pull/8706/commits/465fee0a6a3c77969f77223165f6d7cfb662bec5#diff-8e060ae9fbf3206b4426df775feff178c64694a6c05ae6f34a75a58d302192b5)

File: src/ui/layout/index.md (hunk 4 above)
Comment (rexthecoder) [reply src/ui/layout/index.md:1255]: Can I talk about the dynamic colors over there?
https://pub.dev/packages/dynamic_color

Comment (rexthecoder) [reply src/ui/layout/index.md:1255]: @guidezpl

Comment (guidezpl) [reply src/ui/layout/index.md:1255]: I would remove this example entirely

File: src/get-started/flutter-for/react-native-devs.md [hunk 5]
```diff
@@ -2593,7 +2593,7 @@ and common widget properties.
 [`Axis`]: {{site.api}}/flutter/painting/Axis.html
 [`BuildContext`]: {{site.api}}/flutter/widgets/BuildContext-class.html
 [`Center`]: {{site.api}}/flutter/widgets/Center-class.html
-[color palette]: {{site.material}}/design/color
+[color palette]: {{site.material}}/styles/color
```
Comment (guidezpl) [inline src/get-started/flutter-for/react-native-devs.md:2596]: This should point to the "2014 Material Design color palettes" section under https://m2.material.io/design/color/the-color-system.html#tools-for-picking-colors

**Review by parlough — APPROVED** [review]
Thank you for updating and improving these links!

**Pull request merged**